pkg/controller/machine-set-boot-image/platform_helpers.go

```diff
@@ -192,9 +192,51 @@ func reconcileLibvirt(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap
 	return false, nil, nil
 }
 
-func reconcileVSphere(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
-	klog.Infof("Skipping machineset %s, unsupported platform type VSphere with %s arch", machineSet.Name, arch)
-	return false, nil, nil
+func reconcileVSphere(machineSet *machinev1beta1.MachineSet, infra *osconfigv1.Infrastructure, configMap *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
```

Comment:
What if there are multiple machinesets?

Reply:
this logic is handled in [syncMAPIMachineSets](https://github.com/openshift/machine-config-operator/blob/07ba85e326d5777452d555e179b99309c6aa4a6b/pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go#L373) -> [syncMAPIMachineSet](https://github.com/openshift/machine-config-operator/blob/07ba85e326d5777452d555e179b99309c6aa4a6b/pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go#L461) -> [checkMachineSet](https://github.com/openshift/machine-config-operator/blob/07ba85e326d5777452d555e179b99309c6aa4a6b/pkg/controller/machine-set-boot-image/platform_helpers.go#L91) -> reconcileVSphere